app/src/main/java/com/example/danil/androidcourses/explicitIntent/ExplicitActivity.java

```diff
@@ -0,0 +1,32 @@
+package com.example.danil.androidcourses.explicitIntent;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import com.example.danil.androidcourses.R;
+
+public class ExplicitActivity extends Activity implements View.OnClickListener{
+
+    public final static String KEY = "user";
+    EditText editText;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_explicit_intent);
+        editText = (EditText)findViewById(R.id.text_data);
+        Button button = (Button) findViewById(R.id.btn_next);
```

Comment:
We could simplify it

just use 
((Button) findViewById(R.id.btn_next)).setOnClickListener(this);

instead of

Button button = ((Button) findViewById(R.id.btn_next);
button.setOnClickListener(this);